source/communication_interfaces/include/communication_interfaces/sockets/ZMQPublisherSubscriber.hpp

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include "communication_interfaces/sockets/ISocket.hpp"
+#include "communication_interfaces/sockets/ZMQPublisher.hpp"
+#include "communication_interfaces/sockets/ZMQSubscriber.hpp"
+
+namespace communication_interfaces::sockets {
+
+/**
+ * @brief A class that combines both a ZMQ Publisher and Subscriber socket into one single object
+ */
+class ZMQPublisherSubscriber : public ISocket {
+public:
+  /**
+   * @brief Add the parameters with no or default value to the map
+   * @param context The ZMQ context to be used for the sockets
+   * (it's recommended to only use one context per application)
+   */
+  explicit ZMQPublisherSubscriber(const std::shared_ptr<zmq::context_t>& context);
```

Comment:
I realize that the other ZMQSocket also takes context in the constructor, which is fine by itself. But based on the conversation we had about the parameter map, it gives me the following thoughts:

- the context is a kind of parameter, so it shouldn't really be treated differently to the other parameters 

The context object is not packable in the parameter interface container. But, the context could be an integer (the context ID), and then the context can be constructed by the class. But also:
 
- all socket parameters are very unlikely to change after construction, especially after a socket is opened

So it's fair to say that a generic options struct might well be the simpler alternative to the parameter map and parameter list, in which case it would be possible to pass the context object directly.